packages/samples/headless-react/src/components/pager/pager.fn.tsx

```diff
@@ -0,0 +1,61 @@
+import {useEffect, useState, FunctionComponent} from 'react';
+import {
+  buildPager,
+  Pager as HeadlessPager,
+  PagerOptions,
+} from '@coveo/headless';
+import {engine} from '../../engine';
+
+interface PagerProps {
+  controller: HeadlessPager;
+}
+
+export const Pager: FunctionComponent<PagerProps> = (props) => {
+  const {controller} = props;
+  const [state, setState] = useState(controller.state);
+
+  useEffect(() => controller.subscribe(() => setState(controller.state)), []);
+
+  const previousPage = () => {
+    controller.previousPage();
+  };
+
+  const nextPage = () => {
+    controller.nextPage();
+  };
+
+  const selectPage = (page: number) => {
+    controller.selectPage(page);
+  };
+
+  const isCurrentPage = (page: number) => {
+    return controller.isCurrentPage(page);
+  };
```

Comment:
Same comment as above on removing the functions, and using the controller inside the `return` block.

Ideally, functional React samples would be in the ~30 line range.